backend/routes/auth.py

```diff
@@ -0,0 +1,40 @@
+"""
+Decorator for granting access based on user group
+
+"""
+from base64 import b64decode
+from json import loads
+
+from flask import request
+from werkzeug.exceptions import Unauthorized
+
+
+def auth_group(group):
+    def wrapped(func):
+        def inner(self, *args, **kwargs):
+            if validate_auth(group):
+                return func(self, *args, **kwargs)
+        return inner
+    return wrapped
+
+
+def parse_groups():
+    """
+    Returns list of cognito groups from request
+
+    Note: Still need to implement jwt signature validation
+
+    """
+    oidc_data = request.headers.get("x-amzn-oidc-data")
```

Comment:
It's looking likely we'll end up having to add an `Authorization: Bearer {token}` pass through from the FE/API Gateway (but still debugging). In this case:
 - The token will either be the ID or Access Token (depending on some config I don't understand).
 - If the ID Token, we may have to do extra work to get extra claims?